### ISSUE TITLE
Feature request: be more like readline

### ISSUE THREAD

**Issue by jyn514** [chriswalz/bit#24]
I'd expect the following behavior that isn't currently present:

- ctrl+c should clear the input buffer, not exit completely
- alt+left/alt+right should go back and forward a word. I think this is specific to my .inputrc; in that case the feature request changes to 'read inputrc' :P
- more readline keybindings; in particular `ctrl+o` will run the current command without clearing the input buffer

**Comment by nanohard** [comment]
I fully expect every command line program to exit with ^C, as it is the default SIGINT to exit/break.

**Comment by jyn514** [comment]
I expect command line programs that *I'm* waiting on to exit on ^C - I'm interrupting the computation. I expect command line programs that are waiting on *me* to reset on ^C - I'm interrupting my input.

**Comment by nanohard** [comment]
Ahhhh! Now I get you: my mistake. Yeah, that sounds like a good feature.

**Comment by wookayin** [comment]
+1. Having emacs-like readline keybinding would be great, e.g. CTRL-A, CTRL-E, CTRL-U, etc. Even better if it supports `inputrc`.

**Comment by chriswalz** [comment]
You can now use emacs key binds with 14cb613e5091461df145a233e272357aee904027

running the current command without clearing the buffer is interesting. I'm curious of others would like that. 

In terms of the behavior of ctrl + c. I personally prefer it to exit in this particular case since I think some people may get confused if they can't exit with ctrl + c. Depending on feedback I'm certainly open to changing it.

**Comment by chriswalz** [comment]
I'm gong to consider this issue done. Please file another issue if bit v0.6.1 is missing anything
